.github/workflows/run-test-suite.yml

```diff
@@ -0,0 +1,76 @@
+---
+name: aep
+on:
+  pull_request:
+    branches: 
+      - 'main'
+
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+
+    env:
+      AZURE_VAULT_ID: ${{ secrets.TEST_AZURE_VAULT_ID }}
+      AZURE_CLIENT_ID: ${{ secrets.TEST_AZURE_CLIENT_ID }}
+      AZURE_TENANT_ID: ${{ secrets.TEST_AZURE_TENANT_ID }}
+      AZURE_CLIENT_SECRET: ${{ secrets.TEST_AZURE_CLIENT_SECRET }}
+      AZURE_OPENAI_ENDPOINT: ${{ secrets.TEST_AZURE_OPENAI_ENDPOINT }}
+      AZURE_OPENAI_KEY: ${{ secrets.TEST_AZURE_OPENAI_KEY }}
+      OPENAI_API_KEY: ${{ secrets.TEST_AZURE_OPENAI_KEY }}
+      AZURE_CS_ENDPOINT: ${{ secrets.TEST_AZURE_CS_ENDPOINT }}
+      AZURE_CS_KEY: ${{ secrets.TEST_AZURE_CS_KEY }}
+
+      SYSTEM_PROMPT_FILE: "system_prompts/prompts.json"
+      azure_openai_api_version: "2023-12-01-preview"
+      SYSTEM_API_KEY: "system"
+      OPENAI_API_TYPE: "azure"
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
```

Comment:
Could we use/create a _make unittest_ instead of calling tasks? Would help with aligning to local dev